src/Components/TimePicker/Component.purs

```diff
@@ -0,0 +1,285 @@
+  , search :: String
+  }
+
+type Input =
+  { selection :: Maybe Time
+  }
+
+data Query a
+  = HandleSelect (Select.Message Query TimeUnit) a
+  | Initialize a
+  | TriggerFocus a
+  | Synchronize a
+  | GetSelection (Maybe Time -> a)
+  | SetSelection (Maybe Time) a
+  | Key KeyboardEvent a
+  | Search String a
+  | Receive Input a
+
+data Message
+  = SelectionChanged (Maybe Time)
+  | VisibilityChanged Select.Visibility
+  | Searched String
+
+type ParentHTML m
+  = H.ParentHTML Query ChildQuery Input m
+
+type ChildSlot = Unit
+type ChildQuery = Select.Query Query TimeUnit
+
+----------
+-- Time Units
+
+data TimeUnit
+  = TimeUnit SelectableStatus SelectedStatus Time
+
+data SelectableStatus
+  = NotSelectable
+  | Selectable
+
+data SelectedStatus
+  = NotSelected
+  | Selected
+
+dropdownClasses :: Array HH.ClassName
+dropdownClasses = HH.ClassName <$>
+  [ "absolute"
+  , "shadow"
+  , "max-h-80"
+  , "w-full"
+  , "overflow-y-scroll"
+  , "pin-t"
+  , "pin-l"
+  , "z-50"
+  , "border"
+  , "border-grey-90"
+  , "rounded"
+  , "bg-white"
+  , "text-center"
+  ]
+
+component :: ∀ m
+  . MonadAff m
+ => H.Component HH.HTML Query Input Message m
+component =
+  H.lifecycleParentComponent
+    { initialState
+    , render
+    , eval
+    , receiver: HE.input Receive
+    , initializer: Just $ H.action Initialize
+    , finalizer: Nothing
+    }
+  where
+    initialState :: Input -> State
+    initialState { selection } =
+      { selection
+      , search: ""
+      }
+
+    eval
+      :: Query
+      ~> H.ParentDSL State Query ChildQuery Unit Message m
+    eval = case _ of
+      Search text a -> do
+        H.raise $ Searched text
+        H.liftEffect $ log ("Search: " <> text)
+        case text of
+          "" -> eval $ SetSelection Nothing a
+          _  -> case Utils.guessTime text of
+            Nothing -> pure a
+            Just t  -> do
+              _ <- eval $ SetSelection (Just t) a
+              _ <- H.query unit $ Select.setVisibility Select.Off
+              pure a
+
+      HandleSelect m a -> case m of
+        Select.Emit query -> eval query *> pure a
+
+        Select.Selected (TimeUnit _ _ time) -> do
+          -- We'll want to select the item here, set its status, and raise
+          -- a message about its selection.
+          H.liftEffect $ log ("Selected! Choice was " <> show time)
+          H.modify_ _ { selection = Just time }
+          _ <- H.query unit $ Select.setVisibility Select.Off
+          H.raise $ SelectionChanged $ Just time
+          eval $ Synchronize a
+
+        Select.Searched text -> do
+          H.modify_ _ { search = text }
+          -- we don't actually want to match on search, we want to wait
+          -- until they hit ENTER and then we'll try to match their search
+          pure a
+
+        Select.VisibilityChanged visibility -> do
+          H.raise $ VisibilityChanged visibility
+          pure a
+
+      Initialize a -> do
+        -- TODO: use or delete
```

Comment:
I vote for delete unless it's necessary to set the current time by default or something.